relayer/rpc_relayer.go

```diff
@@ -127,12 +129,44 @@ func (r *RpcRelayer) Relay(ctx context.Context, signedTxs *sequence.SignedTransa
 		return "", nil, nil, err
 	}
 
-	// send to guest module if factory address is the recipient (in order to deploy wallet)
-	// todo: split wallet create and other transactions
-	for _, txn := range signedTxs.Transactions {
-		if txn.To == signedTxs.WalletContext.FactoryAddress {
+	if r.IsDeployTransaction(signedTxs) {
+		to = signedTxs.WalletContext.GuestModuleAddress
+	} else {
+		isDeployed, err := sequence.IsWalletDeployed(r.GetProvider(), to)
```

Comment:
we should add an in-memory cache here.. as testing this each time, is going to be slow and resource intensive

Reply:
This is now done in api waas. So I have removed it from here.